spyder/plugins/base.py

```diff
@@ -11,19 +11,74 @@
 # Third party imports
 import qdarkstyle
 from qtpy.QtCore import Qt, Slot
-from qtpy.QtGui import QKeySequence
-from qtpy.QtWidgets import QDockWidget, QMainWindow, QShortcut
+from qtpy.QtGui import QCursor, QKeySequence
+from qtpy.QtWidgets import (QApplication, QDockWidget, QMainWindow, QMenu,
+                            QMessageBox, QShortcut, QToolButton)
 
 # Local imports
 from spyder.config.base import _
 from spyder.config.gui import is_dark_interface, get_font
 from spyder.config.main import CONF
-from spyder.py3compat import is_text_string
+from spyder.config.user import NoDefault
+from spyder.py3compat import configparser, is_text_string
 from spyder.utils import icon_manager as ima
-from spyder.utils.qthelpers import create_action
+from spyder.utils.qthelpers import (
+    add_actions, create_action, create_toolbutton, MENU_SEPARATOR,
+    toggle_actions)
 from spyder.widgets.dock import SpyderDockWidget
 
 
+class BasePluginMixin(object):
+    """Implementation of the basic functionality for Spyder plugins."""
+
+    def __init__(self, parent=None):
+        super(BasePluginMixin, self).__init__()
+
+        # Assert the plugin has CONF_SECTION
+        assert self.CONF_SECTION is not None
+
+        # Check compatibility
+        check_compatibility, message = self.check_compatibility()
+        if not check_compatibility:
+            self.show_compatibility_message(message)
+
+    def set_option(self, option, value):
+        """Set option in spyder.ini"""
```

Comment:
Period.

Reply:
That's confusing given the period before.

Reply:
Fair enough